lib/matplotlib/offsetbox.py

```diff
@@ -272,23 +272,15 @@ def offset(width, height, xdescent, ydescent, renderer) \
         self._offset = xy
         self.stale = True
 
-    def get_offset(self, width, height, xdescent, ydescent, renderer):
+    def get_offset(self, *args, **kwargs):
```

Comment:
This is deeply confusing. It's almost impossible for the user to figure out whether he has to pass anything and if so what. We can't have that on public API. At the very least, you should make the parameters explicit

```suggestion
    def get_offset(self, width=None, height=None, xdescent=None, ydescent=None, renderer=None):
```

But even with that I'm very concerned with the API. "The method needs arguments depending on the subclass and/or some configurable state." is not a contract you can program against. It's bad enough that we currently have the inconsistency that some subclasses do not take arguments. But at least that's explicit and when you know the class, you know what to pass. This change here would make that more implicit.

Reply:
I still propose to leave the parameters explicit until they are removed completely.

Reply:
The way the API currently is is just as bad though.  In some cases you have to pass arguments .... just for us to drop them on the floor!

I think that this is going the right direction (no inputs anywhere), but I think I agree we need more steps to get there.